module/os/windows/zfs/zfs_debug.c

```diff
@@ -252,7 +252,12 @@ __dprintf(boolean_t dprint, const char *file, const char *func,
 
 	DTRACE_PROBE1(zfs__dbgmsg, char *, zdm->zdm_msg);
 
-	__zfs_dbgmsg(buf);
```

Comment:
Alas, I believe `kmem_alloc()` call  on line 235 will/can use mutex, as well as call into Windows to get more memory, and similarly `kmem_free()`. I am unsure if `vsnprintf()` is safe to call, we'd have to look into that.  So in the end, it might simply be easier to bail.

Reply:
I'm a bit confused, I thought you wanted me to keep the "printBuffer("%s\n", buf);" part working?
Also I'm not sure if it is safe but I didn't have a crash when I tested this this morning.
Do you want me to rollback the latest commit and instead just skip everything?

Reply:
You can keep `printBuffer()` that is safe, but the `buf` wouldn't be generated here, it is done in the `vsnprintf()` above, which uses `kmem_alloc()` to point `buf` to memory. But possibly you can change it into a `printBuffer(fmt, _VA_ARGS_);` or whatever the magic would become

Reply:
Ahh I understand